lib/broccoli/translation-reducer/index.js

```diff
@@ -222,6 +219,11 @@ class TranslationReducer extends CachingWriter {
         encoding: 'utf8',
       });
     }
+
+    // Note: lint after the fallback has been merged in
+    const lintResults = this.linter.lint(translations);
```

Comment:
This will break fallbackLocale since we'll never know if a translation is missing in order for errorOnMissingTranslations to continue to behave as-is.  May need to move this.

I really dislike most of the linting logic and configuration, may end up rewriting it all.